tensorbay/opendataset/RarePlanesSynthetic/loader.py

```diff
@@ -0,0 +1,142 @@
+            masks/
+                Atlanta_Airport_0_0_101_1837_mask.png
+                ...
+            xmls/
+                Atlanta_Airport_0_0_101_1837.xml
+                ...
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+    category_ids = dataset.catalog.panoptic_mask.get_category_to_index()
+    segment = dataset.create_segment()
+    original_mask_dir = os.path.join(root_path, "masks")
+    new_mask_dir = os.path.join(root_path, "new_masks")
+    os.makedirs(new_mask_dir, exist_ok=True)
+    annotation_dir = os.path.join(root_path, "xmls")
+    for image_path in glob(os.path.join(root_path, "images", "*.png")):
+        segment.append(
+            _get_data(image_path, original_mask_dir, annotation_dir, new_mask_dir, category_ids)
+        )
+    return dataset
+
+
+def _get_data(
+    image_path: str,
+    original_mask_dir: str,
+    annotation_dir: str,
+    new_mask_dir: str,
+    category_ids: Dict[str, int],
+) -> Data:
+    stem = os.path.splitext(os.path.basename(image_path))[0]
+    mask_path = os.path.join(new_mask_dir, f"{stem}.png")
+    data = Data(image_path)
+    label = data.label
+    with open(os.path.join(annotation_dir, f"{stem}.xml"), "r", encoding="utf-8") as fp:
+        labels: Any = xmltodict.parse(fp.read())["image"]
+    label.box2d, label.panoptic_mask = _get_box2d_and_panoptic_mask(
+        labels["object"],
+        os.path.join(original_mask_dir, f"{stem}_mask.png"),
+        mask_path,
+        category_ids,
+    )
+    label.classification = _get_classification(labels["JSON_Variation_Parameters"]["parameter"])
+    return data
+
+
+def _get_classification(classification_labels: List[Dict[str, str]]) -> Classification:
+    classification_attributes: Dict[str, Union[int, float, str]] = {}
+    for indexes, attribute_getter in _ATTRIBUTES_GETTER.items():
+        for index in indexes:
+            attribute = classification_labels[index]
+            classification_attributes[attribute["@name"]] = attribute_getter(attribute["@value"])
+    return Classification(attributes=classification_attributes)
+
+
+def _get_box2d_and_panoptic_mask(
+    objects: Any,
+    original_mask_path: str,
+    mask_path: str,
+    category_ids: Dict[str, int],
+) -> Tuple[List[LabeledBox2D], PanopticMask]:
+    all_category_ids: Dict[int, int] = {}
+    original_mask = np.array(Image.open(original_mask_path))
+    mask = np.array(np.zeros(original_mask.shape[:2], dtype=np.uint8))
+    box2ds: List[LabeledBox2D] = []
+    if not isinstance(objects, list):
+        objects = [objects]
+    for index, obj in enumerate(objects, 1):
+        category = ".".join(takewhile(lambda x: x, (obj.get(f"category{i}", "") for i in range(6))))
```

Comment:
what does the `6` come from?

Reply:
Can we use `bool` to replace `lambda x: x`

Reply:
the categories in xml files are up to 6

Reply:
shall we use a named constant for the `6`?